test/contracts/airdrop/pay_airdrop.js

```diff
@@ -0,0 +1,156 @@
+    contracts = await airdrop_utils.deployAirdrop(artifacts, accounts);
+    token     = contracts.token;
+    airdrop   = contracts.airdrop;
+
+    await token.setBalance(airdropBudgetHolder, airdropAmount);
+  });
+
+	it('fails to airdrop and pay if msg.sender is not an active worker', async () => {
+    await airdrop_utils.utils.expectThrow(airdrop.payAirdrop.call(beneficiary, transferAmount, commissionBeneficiary, commissionAmount,
+      '', 0, spender, airdropAmount, { from: opsAddress }));
+  });
+
+	it('fails to airdrop and pay if spender is null', async () => {
+    await airdrop_utils.utils.expectThrow(airdrop.payAirdrop.call(beneficiary, transferAmount, commissionBeneficiary, commissionAmount,
+      '', 0, 0, airdropAmount, { from: worker }));
+  });
+
+	it('fails to airdrop and pay if isValidBeneficiaryData returns false', async () => {
+    // transferAmount == 0
+    await airdrop_utils.utils.expectThrow(airdrop.payAirdrop.call(beneficiary, 0, commissionBeneficiary, commissionAmount,
+      '', 0, spender, airdropAmount, { from: worker }));
+  });
+
+  it('fails to airdrop and pay if unable to perfrom airdrop transfer to spender', async () => {
+    await airdrop_utils.utils.expectThrow(airdrop.payAirdrop.call(beneficiary, transferAmount, commissionBeneficiary, commissionAmount,
+      '', 0, spender, airdropAmount, { from: worker }));
+  });
+
+  it('fails to airdrop and pay if unable to perform transfers', async () => {
+    await token.approve(airdrop.address, airdropAmount, { from: airdropBudgetHolder });
+    await airdrop_utils.utils.expectThrow(airdrop.payAirdrop.call(beneficiary, transferAmount, commissionBeneficiary, commissionAmount,
+      '', 0, spender, airdropAmount, { from: worker }));
+  });
+
+  it('successfully airdrops and pays', async () => {
+    await token.approve(airdrop.address, transferAmount.plus(commissionAmount), { from: spender });
+    var totalPaid = await airdrop.payAirdrop.call(beneficiary, transferAmount, commissionBeneficiary, commissionAmount,
+      '', 0, spender, airdropAmount, { from: worker });
+    assert.equal((totalPaid).toNumber(), transferAmount.plus(commissionAmount).toNumber());
+
+    response = await airdrop.payAirdrop(beneficiary, transferAmount, commissionBeneficiary, commissionAmount,
+      '', 0, spender, airdropAmount, { from: worker });
+    assert.equal(await token.balanceOf.call(spender), 0);
+    // Airdrop only uses as much of the airdrop amount as required to cover total paid
+    assert.equal((await token.allowance.call(airdropBudgetHolder, airdrop.address)).toNumber(), airdropAmount.minus(totalPaid).toNumber());
+    assert.equal((await token.balanceOf.call(beneficiary)).toNumber(), transferAmount.toNumber());
+    assert.equal((await token.balanceOf.call(commissionBeneficiary)).toNumber(), commissionAmount.toNumber());
+    checkAirdropPaymentEvent(response.logs[0], beneficiary, transferAmount, commissionBeneficiary, commissionAmount, '', 0, spender, airdropAmount);
```

Comment:
As per new AirdropPayment event parameters
transferAmount => tokenAmount
commissionAmount => commissionTokenAmount

Reply:
Should be validated with call method output

Reply:
if needed we can add tokenAmount, commissionTokenAmount in payAirdrop return.

Reply:
@abhayks1: here is *without currency*. See below that in the test for successfully airdropping with currency, I call `getPricePointAndCalculatedAmounts` to set the `tokenAmount` and `commissionTokenAmount` vars; this difference in the tests/vars was intentional, in the hope of being instructive of what happens in the currency case. Please let me know if you think it's better to set the amounts to token-amount-named vars.

cc @benjaminbollen